### PR TITLE
API-4410: Name change topic to box. Also changed notifications routes…

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
… to be box/boxId/notifications

### CODE REVIEW

File: it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala [hunk 1]
```diff
@@ -85,92 +85,92 @@ class NotificationsControllerISpec extends ServerBaseISpec with BeforeAndAfterEa
       .futureValue
   // need to clean down mongo then run two
 
-  def createTopicAndReturn(): Topic = {
-    val result = doPut( s"$url/topics", createTopicJsonBody, validHeadersJson)
+  def createBoxAndReturn(): Box = {
+    val result = doPut( s"$url/box", createTopicJsonBody, validHeadersJson)
```
Comment (lburgos) [inline it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala:89]: topic

File: it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala [hunk 2]
```diff
@@ -43,10 +43,10 @@ class NotificationsControllerISpec extends ServerBaseISpec with BeforeAndAfterEa
 
   val url = s"http://localhost:$port"
 
-  val topicName = "mytopicName"
+  val boxName = "myboxName"
   val clientId = "someClientId"
-  val createTopicJsonBody =raw"""{"clientId": "$clientId", "topicName": "$topicName"}"""
-  val createTopic2JsonBody =raw"""{"clientId": "zzzzzzzzzz", "topicName": "bbyybybyb"}"""
+  val createTopicJsonBody =raw"""{"clientId": "$clientId", "boxName": "$boxName"}"""
+  val createTopic2JsonBody =raw"""{"clientId": "zzzzzzzzzz", "boxName": "bbyybybyb"}"""
```
Comment (lburgos) [inline it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala:49]: topic

File: app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala [hunk 3]
```diff
@@ -45,25 +45,25 @@ object ResponseFormatters{
...
   implicit val createNotificationResponseFormatter: OFormat[CreateNotificationResponse] = Json.format[CreateNotificationResponse]
 }
 
 object RequestFormatters {
   implicit val clientIdFormatter: Format[ClientId] = Json.valueFormat[ClientId]
-  implicit val topicIdFormatter: Format[TopicId] = Json.valueFormat[TopicId]
+  implicit val boxIdFormatter: Format[BoxId] = Json.valueFormat[BoxId]
   implicit val subscriberIdFormatter: Format[SubscriberId] = Json.valueFormat[SubscriberId]
-  implicit val createTopicRequestFormatter: OFormat[CreateTopicRequest] = Json.format[CreateTopicRequest]
+  implicit val createTopicRequestFormatter: OFormat[CreateBoxRequest] = Json.format[CreateBoxRequest]
```
Comment (lburgos) [inline app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala:66]: topic

File: app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala [hunk 4]
```diff
@@ -45,25 +45,25 @@ object ResponseFormatters{
...
   implicit val subscriberIdFormatter: Format[SubscriberId] = Json.valueFormat[SubscriberId]
   implicit val pushSubscriberFormats: OFormat[PushSubscriber] = Json.format[PushSubscriber]
   implicit val formatSubscriber: Format[Subscriber] = Union.from[Subscriber]("subscriptionType")
     .and[PushSubscriber](SubscriptionType.API_PUSH_SUBSCRIBER.toString)
     .format
-  implicit val topicsFormats: OFormat[Topic] = Json.format[Topic]
+  implicit val boxFormats: OFormat[Box] = Json.format[Box]
   implicit val notificationFormatter: OFormat[Notification] = Json.format[Notification]
-  implicit val createTopicResponseFormatter: OFormat[CreateTopicResponse] = Json.format[CreateTopicResponse]
+  implicit val createTopicResponseFormatter: OFormat[CreateBoxResponse] = Json.format[CreateBoxResponse]
```
Comment (lburgos) [inline app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala:58]: topic

File: it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala:89]: doh!!

File: it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply it/uk/gov/hmrc/pushpullnotificationsapi/controllers/NotificationsControllerISpec.scala:49]: doh!! will fix

File: app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala:66]: will fix!

File: app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala (hunk 4 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsapi/models/jsonformatters.scala:58]: Will Fix

**Pull request merged**